Package IndexRange for netStandard2.0 projects can cause conflicts with Microsoft.Bcl.Memory

### Observed behavior

Dependency conflict between `IndexRange`and `Microsoft.Bcl.Memory` can cause duplicate `System.Index` definitions when using NATS from netframework/netstandard2.0. (Same issue for `Range`)
 
* `Nats.Client.Core` has a dependency to `IndexRange` in version 1.0.3
*  The project using NATS has a (transitive) dependency to `Microsoft.Bcl.Memory`
*  Both packages provide the C#8 Range and Index-Syntax
*  This will give compile time erros, as both packages define the sames types.

```
# Error Example
path-to-file-with-error.cs(420,69): error CS8356: Predefined type 'System.Index' is declared in multiple referenced assemblies:
'IndexRange, Version=1.0.3.0, Culture=neutral, PublicKeyToken=35e6a3c4212514c6' and
'Microsoft.Bcl.Memory, Version=9.0.0.8, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'
```

### Expected behavior

NATS does not include the `IndexRange`-Packge.

I recommend removing this package entirely, as it can be replaced by the Microsoft one.
This is the suggested cause to action by the maintainer.
See [issue 14 in the IndexRange Repo](https://github.com/bgrainger/IndexRange/issues/14) and the Repo-Readme

> [IndexRange Repo readme](https://github.com/bgrainger/IndexRange/blob/master/README.md#indexrange)
For projects that target netstandard2.0 or .NET Framework 4.6.2 or later, use the [Microsoft.Bcl.Memory](https://www.nuget.org/packages/Microsoft.Bcl.Memory/) package instead. This package should be considered deprecated for those target frameworks.

I created the draft #1009 to show the code-impact of these changes. It's just a 1:1 repalcement of the packages.

### Server and client version

It's a compile time issue with the current released package: 2.6.11 

### Host environment

_No response_

### Steps to reproduce

Use a netFramework project that references NATS (eg. `Nats.Client.Core`) and any package that includes `Microsoft.Bcl.Memory`

--- a/src/NATS.Client.Core/NATS.Client.Core.csproj
+++ b/src/NATS.Client.Core/NATS.Client.Core.csproj
@@ -32,8 +32,8 @@
     <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
         <PackageReference Include="System.Memory" Version="4.5.5"/>
         <PackageReference Include="System.Buffers" Version="4.5.1"/>
-        <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" Version="8.0.0"/>
-        <PackageReference Include="Microsoft.Bcl.Memory" Version="9.0.11"/>
+        <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" Version="9.0.0"/>
+        <PackageReference Include="Microsoft.Bcl.Memory" Version="9.0.0"/>
         <!--  Nerdbank.Streams for SequenceReader support https://github.com/dotnet/standard/issues/1493  -->
         <PackageReference Include="Nerdbank.Streams" Version="2.10.72"/>
     </ItemGroup>

--- a/src/NATS.Client.Core/NATS.Client.Core.csproj
+++ b/src/NATS.Client.Core/NATS.Client.Core.csproj
@@ -32,7 +32,7 @@
     <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
         <PackageReference Include="System.Memory" Version="4.5.5"/>
         <PackageReference Include="System.Buffers" Version="4.5.1"/>
-        <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" Version="9.0.0"/>
+        <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" Version="8.0.0"/>
         <PackageReference Include="Microsoft.Bcl.Memory" Version="9.0.0"/>
         <!--  Nerdbank.Streams for SequenceReader support https://github.com/dotnet/standard/issues/1493  -->
         <PackageReference Include="Nerdbank.Streams" Version="2.10.72"/>

--- a/src/NATS.Client.Core/NATS.Client.Core.csproj
+++ b/src/NATS.Client.Core/NATS.Client.Core.csproj
@@ -33,9 +33,9 @@
         <PackageReference Include="System.Memory" Version="4.5.5"/>
         <PackageReference Include="System.Buffers" Version="4.5.1"/>
         <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" Version="8.0.0"/>
+        <PackageReference Include="Microsoft.Bcl.Memory" Version="9.0.11"/>
         <!--  Nerdbank.Streams for SequenceReader support https://github.com/dotnet/standard/issues/1493  -->
         <PackageReference Include="Nerdbank.Streams" Version="2.10.72"/>
-        <PackageReference Include="IndexRange" Version="1.0.3"/>
     </ItemGroup>
 
     <!--  Dependencies for net6.0 and net8.0 only -->

--- a/tests/NATS.Client.TestUtilities/NATS.Client.TestUtilities.csproj
+++ b/tests/NATS.Client.TestUtilities/NATS.Client.TestUtilities.csproj
@@ -20,7 +20,7 @@
   <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
     <PackageReference Include="System.Memory" Version="4.5.5"/>
     <PackageReference Include="System.Buffers" Version="4.5.1"/>
-    <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" Version="9.0.0"/>
+    <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" Version="8.0.0"/>
     <PackageReference Include="Microsoft.Bcl.Memory" Version="9.0.0"/>
     <PackageReference Include="System.Text.Json" Version="8.0.5"/>
     <PackageReference Include="System.Threading.Channels" Version="8.0.0"/>

--- a/tests/NATS.Client.TestUtilities/NATS.Client.TestUtilities.csproj
+++ b/tests/NATS.Client.TestUtilities/NATS.Client.TestUtilities.csproj
@@ -20,8 +20,8 @@
   <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
     <PackageReference Include="System.Memory" Version="4.5.5"/>
     <PackageReference Include="System.Buffers" Version="4.5.1"/>
-    <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" Version="8.0.0"/>
-    <PackageReference Include="Microsoft.Bcl.Memory" Version="9.0.11"/>
+    <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" Version="9.0.0"/>
+    <PackageReference Include="Microsoft.Bcl.Memory" Version="9.0.0"/>
     <PackageReference Include="System.Text.Json" Version="8.0.5"/>
     <PackageReference Include="System.Threading.Channels" Version="8.0.0"/>
     <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="8.0.1"/>

--- a/tests/NATS.Client.TestUtilities/NATS.Client.TestUtilities.csproj
+++ b/tests/NATS.Client.TestUtilities/NATS.Client.TestUtilities.csproj
@@ -21,7 +21,7 @@
     <PackageReference Include="System.Memory" Version="4.5.5"/>
     <PackageReference Include="System.Buffers" Version="4.5.1"/>
     <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" Version="8.0.0"/>
-    <PackageReference Include="IndexRange" Version="1.0.3"/>
+    <PackageReference Include="Microsoft.Bcl.Memory" Version="9.0.11"/>
     <PackageReference Include="System.Text.Json" Version="8.0.5"/>
     <PackageReference Include="System.Threading.Channels" Version="8.0.0"/>
     <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="8.0.1"/>

